rosco-core/src/test/groovy/com/netflix/spinnaker/rosco/providers/aws/AWSBakeHandlerSpec.groovy

```diff
@@ -354,6 +358,71 @@ class AWSBakeHandlerSpec extends Specification {
       1 * packerCommandFactoryMock.buildPackerCommand("", parameterMap, "$configDir/$awsBakeryDefaults.templateFile")
   }
 
+  void 'produces packer command with all required parameters for amzn, with sudo using runAsRoot:true'() {
+    setup:
```

Comment:
On second thought, I am wondering if you need this test at all.
If you can show in other tests that you are calling baseCommand then you just need to add tests that show baseCommand return sudo when you want it to and not when you dont. Then the LocalJobFriendlyPackerCommandFactorySpec below shows that you are constructing the packer command correct. That gives you a more concise set of tests and eliminates a lot of the noise here with all the extra parameter decoration and tedious mocking.

1) you get sudo exactly when you want
2) you build command lines properly with and without sudo
3) your flow includes the calls to determine when you want sudo and will pass that decision into building a command (and uses all the other configuration stuff).

What do you think?